pysph/tools/tests/test_geometry.py

```diff
@@ -104,98 +100,151 @@ def test_get_2d_wall(self):
+        assert len_y == pytest.approx(height)
+        assert len_z == pytest.approx(depth)
         assert np.allclose(center, new_center)
 
     def test_get_3d_hollow_cylinder(self):
-        dx = 0.17
-        radius = np.random.randint(20, 40) * dx
-        length = np.random.randint(20, 40) * dx
+        dx = 0.15
+        radius = 2.0
+        length = 3.0
         center = np.random.random_sample(3)
         num_layers = np.random.randint(1, 5)
-        x, y, z = get_3d_hollow_cylinder(dx, radius, length, center,
-                                         num_layers)
-        x = x - center[0]
-        y = y - center[1]
-        z = z - center[2]
+        x, y, z = G.get_3d_hollow_cylinder(dx, radius, length, center,
+                                           num_layers)
+        xc, yc, zc = x - center[0], y - center[1], z - center[2]
+        assert not np.any(xc * xc + yc * yc) > radius * radius
+        assert not np.any(zc) > (length / 2.0)
+
+    def test_get_4digit_naca_airfoil(self):
+        dx = 0.05
+        c = 1.5
+        camber = '44'
+        t = 10
+        airfoil = np.random.choice(['00', camber]) + str(t)
+        x, y = G.get_4digit_naca_airfoil(dx, airfoil, c)
         count = 0
-        for i in range(len(x)):
-            point = np.array([x[i], y[i]])
-            dist = distance_2d(point)
-            condition_1 = dist >= radius * (1.0 + dx * 1.0e-03)
-            condition_2 = abs(z[i]) >= (length / 2.0) * (1.0 + dx * 1.0e-03)
-            if condition_1 or condition_2:
-                count += 1
+        t = 0.01 * t
+        for xi, yi in zip(x, y):
+            yt = 5.0 * t * (0.2969 * np.sqrt(xi / c) - 0.1260 * (xi / c) -
+                            0.3516 * ((xi / c)**2.) + 0.2843 * (xi / c)**3. -
+                            0.1015 * ((xi / c)**4.))
+            if airfoil[:2] == '00':
+                if abs(yi) > yt + dx:
+                    count += 1
+            else:
+                m = 0.01 * float(airfoil[0])
+                p = 0.1 * float(airfoil[1])
+                if xi <= p * c:
+                    yc = (m / (p * p)) * (2. * p * (xi / c) - (xi / c)**2.)
+                    dydx = (2. * m / (p * p)) * (p - xi / c) / c
+                else:
+                    yc = (m / ((1. - p) * (1. - p))) * \
+                        (1. - 2. * p + 2. * p * (xi / c) - (xi / c)**2.)
+                    dydx = (2. * m / ((1. - p) * (1. - p))) * (p - xi / c) / c
+                theta = np.arctan(dydx)
+                if yi >= 0.0:
+                    yu = yc + yt * np.cos(theta)
+                    if yi > yu + dx:
+                        count += 1
+                else:
+                    yl = yc - yt * np.cos(theta)
+                    if yi < yl + dx:
+                        count += 1
+        assert count == 0
+
+    def test_get_5digit_naca_airfoil(self):
+        dx = 0.05
+        c = 1.5
+        series = np.random.choice(['210', '220', '230', '240', '250', '221',
```

Comment:
Do not make this random.  If you want to test them all, then test them all using a parameter or put it in a for loop.  No sense in just testing one of them each time.  The idea of a test is to cover the possible options but just enough so as to make sure everything is well tested.  You should really measure coverage of the geometry module and make sure that it is almost entirely covered.